lib/compat/wordpress-5.9/block-template-utils.php

```diff
@@ -21,6 +21,36 @@
 	define( 'WP_TEMPLATE_PART_AREA_UNCATEGORIZED', 'uncategorized' );
 }
 
+
+if ( ! function_exists( 'get_block_theme_folders' ) ) {
+	/**
+	 * For backward compatibility reasons,
+	 * block themes might be using block-templates or block-template-parts,
+	 * this function ensures we fallback to these folders properly.
+	 *
+	 * @param string $theme_stylesheet The stylesheet. Default is to leverage the main theme root.
+	 *
+	 * @return array Folder names used by block themes.
+	 */
+	function get_block_theme_folders( $theme_stylesheet = null ) {
```

Comment:
Why not accept `theme_dir` directly and remove calculation logic?

Reply:
From an API point of view, I felt that accepting a them stylesheet (or id) is a bit better.

Reply:
No strong preference. I was just curious :)